File: Question-Two/weather_forecast.py
class WeatherForecastApp:
    def __init__(self):
        self.cities = {}

    def add_city(self, city_name, temperature, humidity, wind_speed):
        self.cities[city_name] = {
            "temperature":temperature,
            "humidity":humidity,
            "wind_speed":wind_speed
        }
        return f"Added weather data for {city_name}: Temp = {temperature} degrees celcius, Humidity = {humidity}%, Wind = {wind_speed}km/h"

    def get_city_weather(self,city_name):
        info = self.cities.get(city_name)
        if info:
            print(f"{city_name} Weather =>Temp:{info['temperature']} degrees celcius, Humidity:{info['humidity']}%, Wind:{info['wind_speed']}km/h")
            return info
        else:
            print(f"No weather data found for {city_name}")
        
        

File: Question-Two/weather-forecast.kt
fun main() {
   
 val weatherApp = WeatherForecastApp()


    weatherApp.addCity("Nairobi", 22.5f, 60, 5.0f)
    weatherApp.addCity("Mombasa", 25.0f, 50, 3.5f)
    weatherApp.addCity("Kisumu", 18.0f, 70, 4.0f)

    weatherApp.WeatherData()
}

data class WeatherInfo(
    val cityName: String,
    var temperature: Float,
    var humidity: Int,
    var windSpeed: Float
)
class WeatherForecastApp {
    val weatherData: MutableList<WeatherInfo> = mutableListOf()

   
    fun addCity(cityName: String, temperature: Float, humidity: Int, windSpeed: Float) {
        val newCityWeather = WeatherInfo(cityName, temperature, humidity, windSpeed)
        weatherData.add(newCityWeather)
        println("Added weather data for city: $cityName")
    }


    fun WeatherData() {
        for (weather in weatherData) {
            println("City: ${weather.cityName}, Temperature: ${weather.temperature}degrees celsius, Humidity: ${weather.humidity}%, Wind Speed: ${weather.windSpeed} m/s")
        }
    }
}